src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2513,6 +2515,360 @@ public Expand(Func<TIn, IEnumerator<TOut>> extrapolate)
         public override string ToString() => "Expand";
     }
 
+    public sealed class NotYetThereSentinel : Exception
+    {
+        public static readonly NotYetThereSentinel Instance = new();
+    }
+
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <typeparam name="TIn">TBD</typeparam>
+    /// <typeparam name="TOut">TBD</typeparam>
+    [InternalApi]
+    public sealed class
+        SelectValueTaskAsync<TIn, TOut> : GraphStage<FlowShape<TIn, TOut>>
```

Comment:
Same nitpicks about naming

Reply:
1.5 points here:

1. The class itself, will almost certainly require a different name from the original
1.5.  As for the DSL, Will see whether having the same exposed name causes issues; in _this_ (i.e. as opposed to `UnfoldResource`) case I'm more worried about compiler being confused as to which overload to use and diminish dev experience.

Reply:
Per discord comments giving the DSL methods the same name results in compiler confusion due to inability to auto-resolve inlined async Func, Happy to give DSL a better name though and open to options. :)